Tests/UnitTests/Diagnostics/DiagnosticsSynchronizerTests.swift

```diff
@@ -177,6 +177,19 @@ class DiagnosticsSynchronizerTests: TestCase {
         expect(self.userDefaults.removeObjectForKeyCalledValues) == [cacheKey]
     }
 
+    func testSyncMultipleEventsWithInvalidEvent() async throws {
```

Comment:
Should we add a test with invalid events but that leaves the expected amount in disk? Don't remember if that was easy to do though 😅

Reply:
Not really sure what you mean @tonidero 🤔

Reply:
Right sorry 😅, meant a test that has invalid events, syncs the events, but the events in the cache are still bigger than the number of events synced, so it leaves a few behind, possibly also some invalid events

Reply:
If I'm not mistaken, when a sync happens, all the events in the file get synced.
So the situation you mention could only happen if a new event gets added in the middle of a sync (after the entries have been read and before the synced entries have been deleted). Is that the scenario you mean?

Reply:
Ohhh ok... In Android, we do have a maximum number of events that can be synced per request. But yeah you're right, we don't have that in iOS. So yeah, I think the test I was proposing doesn't make sense, sorry about the confussion 🙇

Reply:
No problem. Thanks for the review!